lldp_util.c

```diff
@@ -1287,6 +1287,7 @@ int get_arg_val_list(char *ibuf, int ilen, int *ioff,
 			p = (int *) realloc(argvallens,
 				(i/NUM_ARGS + 1) * NUM_ARGS * sizeof(int));
 			if (!p) {
+				free(arglens);
```

Comment:
This is good, but it also highlights another possible memory leak. I suspect the `return` just above on line 1282 also needs a call to `free(argvallens)` due to how this loop operates repeatedly calling `realloc()` as needed to ensure there's enough space. It is probably better to instead have the two `free()` calls in the function epilogue and just `numargs = 0; goto out;`. Both `arglens` and `argvallens` will be either `NULL` or the last successfully allocated memory block so it's probably cleanest to clean them both up at the end.